packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1600,7 +1605,7 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     super.initState();
     _clipboardStatus?.addListener(_onChangedClipboardStatus);
     widget.controller.addListener(_didChangeTextEditingValue);
-    _focusAttachment = widget.focusNode.attach(context);
+    _focusAttachment = widget.focusNode.attach(context, contextBounds: widget.parentContext);
```

Comment:
It looks like the hittest uses prioritize context bounds to determine hittest rect, I wonder can it directly attach to the widget.parentContext if it is not null?

Reply:
Thinking about this more, we might have some issues if editable text migrates to use a regular focus node since we won't have the attachment available.

Attaching to the parent context might work, but thinking about it more I'm not sure if we need to worry about reparenting. Surely you'd never re-parent the text field decoration apart from an editable text right?

Reply:
third party libraries does crazy thing around editable text, I wouldn't be surprised if the editabletext gets reparented. but I think we should be able to reparent the attachment in didUpdateWidget?

Reply:
Currently focus will be lost (at least according to my test cases)

Reply:
yes, focusAttachment.detach will call unfocus on the focusnode. I think this is probably fine because this will be an edgy case where people keep swapping out the parent focus. We can fix it later if that actually cause issue.